### DISCUSSION TITLE
xcodebuild adjusts theme colors in editor

### DISCUSSION THREAD

**Discussion by Madmen777** [wojciech-kulik/xcodebuild.nvim#125] [Q&A]
hi, thank you for this nice nvim plugin. I have an issue and I am not sure how to solve this. when activating the xcodeplugin my selected theme in vim seems to be modified. keywords are displayed in other colors, and I am not sure how I can deactivate this behavior. looked in the documentation and did not find any hints on this. can you help please?

**Comment by wojciech-kulik** [comment]
Probably you haven't got treesitter and you installed it together with xcodebuild.nvim. The plugin itself doesn't change highlighting. In general, highlighting with treesitter is a good thing, basic highlighting is very poor. If you don't want to use it, just remove it from dependencies and uninstall the plugin. It is only required if you want to have support for Quick framework in tests.

**Comment by Woit** [comment]
@Madmen777 i think you should use TreeSitter anyway, because many other very useful plugins use it. If you using some colorscheme which is supporting TreeSitter, you can redefine colors for types/classes/variables/etc. specially for swift files

as example, im using Everforest colorscheme, and did all booleans **bold**
```lua
-- somewhere in everforest.lua config
on_highlights = function(hl, palette)
  hl.TSBoolean = { fg = palette.purple, bg = palette.none, bold = true }
end,
```

**Comment by Madmen777** [comment]
thanks for the answers. yes it is indeed tree sitter as it seems. I have now overwritten the highlight group @type.swift with a vim command (_vim.api.nvim_set_hl(0, "@type.swift", { fg = "#c9770a", bg = "", italic = false, underline = false, bold = false, sp = ""})_) in the init.lua script, and the ugly green highlight of swift types is gone.
